extensions/typescript/src/features/jsDocCompletionProvider.ts

```diff
@@ -0,0 +1,184 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import { Position, Selection, Range, CompletionItemProvider, CompletionItemKind, TextDocument, CancellationToken, CompletionItem, window, commands, Uri, ProviderResult, TextEditor } from 'vscode';
+
+import { ITypescriptServiceClient } from '../typescriptService';
+import { FileLocationRequestArgs, DocCommandTemplateResponse } from '../protocol';
+
+import * as nls from 'vscode-nls';
+const localize = nls.loadMessageBundle();
+
+const tryCompleteJsDocCommand = '_typeScript.tryCompleteJsDoc';
+
+
+class JsDocCompletionItem extends CompletionItem {
+	constructor(file: Uri, position: Position) {
+		super('/** @param */', CompletionItemKind.Snippet);
+		this.detail = localize('typescript.jsDocCompletionItem.detail', 'Complete JSDoc comment');
+		this.insertText = '';
+		this.command = {
+			title: 'Try Complete Js Doc',
+			command: tryCompleteJsDocCommand,
+			arguments: [file, position]
+		};
+	}
+}
+
+export default class JsDocCompletionHelper implements CompletionItemProvider {
+
+	constructor(
+		private client: ITypescriptServiceClient,
+	) {
+		window.onDidChangeTextEditorSelection(e => {
+			if (e.textEditor.document.languageId !== 'typescript'
+				&& e.textEditor.document.languageId !== 'typescriptreact'
+				&& e.textEditor.document.languageId !== 'javascript'
+				&& e.textEditor.document.languageId !== 'javascriptreact'
+			) {
+				return;
+			}
+
+			const selection = e.selections[0];
+			if (!selection.start.isEqual(selection.end)) {
+				return;
+			}
+			if (this.shouldAutoShowJsDocSuggestion(e.textEditor.document, selection.start)) {
+				return commands.executeCommand('editor.action.triggerSuggest');
+			}
+			return;
+		});
+
+		commands.registerCommand(
+			tryCompleteJsDocCommand,
+			(file: Uri, position: Position) => this.tryCompleteJsDoc(file, position));
+	}
+
+	public provideCompletionItems(document: TextDocument, position: Position, _token: CancellationToken): ProviderResult<CompletionItem[]> {
+		const file = this.client.normalizePath(document.uri);
+		if (file) {
+			return [new JsDocCompletionItem(document.uri, position)];
```

Comment:
Unconditionally returning a completion is causing https://github.com/Microsoft/vscode/issues/21147. Expect this method to be called 'randomly' (see above). Because you cannot control when that happens, the function must check the document/postion for a valid jsdoc-insertion locations and return something falsy else